docs/resources/environment_template.md

```diff
@@ -0,0 +1,714 @@
+            success_condition = "200 OK"
+          }
+        }
+        timeout_seconds = 1000
+      }
+    }
+    agents {
+      name = var.agent_name
+    }
+    contexts {
+      name = var.configcontext_name
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `metadata` (Block List, Max: 1) Metadata of the environment template resource (see [below for nested schema](#nestedblock--metadata))
+- `spec` (Block List, Max: 1) Specification of the environment template resource (see [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+***Required***
+
+- `name` (String) name of the resource
+- `project` (String) Project of the resource
+
+***Optional***
+
+- `description` (String) description of the resource
+
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+***Required***
+
+- `resources` (Block List) Specify all the environment resources to be included in environment (see [below for nested schema](#nestedblock--spec--resources))
+- `version` (String) Environment template version
+
+***Optional***
+
+- `agents` (Block List) Agents that are eligible to process the template (see [below for nested schema](#nestedblock--spec--agents))
+- `contexts` (Block List) Reference to config context data associated with environment templates (see [below for nested schema](#nestedblock--spec--contexts))
+- `hooks` (Block List, Max: 1) Specify lifecycle hook actions (see [below for nested schema](#nestedblock--spec--hooks))
+- `sharing` (Block List, Max: 1) Sharing options with other projects (see [below for nested schema](#nestedblock--spec--sharing))
+- `variables` (Block List) Environment variables, file data and other variables (see [below for nested schema](#nestedblock--spec--variables))
+- `version_state` (String) Represents the current state of template version
+
+<a id="nestedblock--spec--agents"></a>
+### Nested Schema for `spec.agents`
+
+***Required***
+
+- `name` (String) name of the resource
+
+
+<a id="nestedblock--spec--contexts"></a>
+### Nested Schema for `spec.contexts`
+
+***Required***
+
+- `name` (String) name of the resource
+
+
+<a id="nestedblock--spec--hooks"></a>
+### Nested Schema for `spec.hooks`
+
+***Optional***
+
+- `on_completion` (Block List) On completion environment hook (see [below for nested schema](#nestedblock--spec--hooks--on_completion))
+- `on_failure` (Block List) On failure environment hook (see [below for nested schema](#nestedblock--spec--hooks--on_failure))
+- `on_init` (Block List) On initialize environment hook (see [below for nested schema](#nestedblock--spec--hooks--on_init))
+- `on_success` (Block List) On success environment hook (see [below for nested schema](#nestedblock--spec--hooks--on_success))
+
+<a id="nestedblock--spec--hooks--on_completion"></a>
```

Comment:
since options are same for on_completion, on_failure, on_init, on_success, instead of duplicating the content, lets add like this  Nested Schema for `spec.hooks.on_completion`, `spec.hooks.on_failure`, `spec.hooks.on_init`, `spec.hooks.on_success` . Similarly for other nested blocks